cloudcix_primitives/storage_kvm.py

```diff
@@ -124,37 +138,38 @@ def update(
     """
     # Define message
     messages = {
-        1000: f'1000: Successfully update storage {storage}',
-        3021: f'3021: Failed to connect to the host {host}',
-        3022: f'3022: Failed to update storage_kvm {domain_path}{storage}'
+        1300: f'1300: Successfully updated storage image {storage} to {size}GB at {domain_path}{storage}'
+              f' on Host {host}.',
+        3321: f'3321: Failed to connect to the Host {host} for payload resize_storage_file.',
+        3322: f'3322: Failed to update storage image {domain_path}{storage} to {size}GB on Host {host}.'
     }
-    message_list = {}
-    # Define payload
-    payload = f'qemu-img resize {domain_path}{storage} {size}G'
-
-    # Update storage using SSH communication
-    response = comms_ssh(
-        host_ip=host,
-        payload=payload,
-        username='robot',
-    )
-    if response['channel_code'] != CHANNEL_SUCCESS:
-        msg = f'{messages[3021]}\n ' \
-                f'channel_code: {response["channel_code"]}s.\n' \
-                f'channel_message: {response["channel_message"]}\n' \
-                f'channel_error: {response["channel_error"]}'
-        message_list.append(msg)
-        success = False
-    if response['payload_code'] != SUCCESS_CODE:
-        msg = f'{messages[3022]}\n ' \
-                f'payload_code: {response["payload_code"]}s.\n' \
-                f'payload_message: {response["payload_message"]}\n' \
-                f'payload_error: {response["payload_error"]}'
-        message_list.append(msg)
-        success = False
-
-    message_list.append(messages[1000])
-    return success, message_list
+
+    def run_host(host, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, host, 'robot')
+        fmt = HostErrorFormatter(
+            host,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads
+        )
+
+        payloads = {
```

Comment:
Is error checking required for if the current size of the storage drive is 100GB and the size in the request is 50GB?

1. Will the `qemu-img resize` fail and return an error? 

2. Will the `qemu-img resize` command be successful and reduce the size of the storage drive?

Reply:
testing how it works

Reply:
`resize` doesn't support reducing the size:
```
(navenv) administrator@dev-pod:~/naveen/primitives/tools$ python test_storage_kvm_primitive.py update
Status: False

Message:
3322: Failed to update storage image /var/lib/libvirt/images/123_234_HDD_568.img to 10GB on Host xxxxxxxxxx
Payload response from the host: xxxxxxxxxxx
payload code: 1
STDERR: qemu-img: warning: Shrinking an image will delete all data beyond the shrunken image's end. Before performing such an operation, make sure there is no important data there.
qemu-img: Use the --shrink option to perform a shrink operation.

STDOUT: 

```

Reply:
Okay, that's good to know it will be prevented and the response gives the reason why.